torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -132,7 +132,90 @@ FunctionParameter::FunctionParameter(const std::string& fmt, bool keyword_only)
   }
 }
 
-bool FunctionParameter::check(PyObject* obj) {
+static PyObject* get_tensor_torch_function(void)
+{
+  PyObject* method = PyObject_GetAttrString((PyObject*)THPVariableClass, "__torch_function__");
+  assert(method != nullptr);
+  return method;
+}
+
+// checks if `overloaded_args[]` has args with `__torch_function__`
+static bool check_has_torch_function(PyObject* obj)
+{
+  PyObject* method = PyTorch_LookupSpecial(obj, "__torch_function__");
+  if(method != nullptr){
+    return true;
+  }
+  return false;
+}
+
+bool FunctionParameter::check_exact(PyObject* obj)
```

Comment:
Since this copies a lot of code from `FunctionParameter::check` it would be quite good to say what the distinction between the two functions are, so future people who need to make modifications know how to apply the changes

Reply:
Agree. I will add the comments.